### DISCUSSION TITLE
Official release of clangd doesn't work with ecode

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#312] [Q&A] [answered]
I switched from FreeBSD to Linux. The distro's version of clangd is too old so I decided to download the official release.

https://github.com/clangd/clangd/discussions/2118

**Comment by SpartanJ** [comment]
I'm not sure what to say, you must provide more information. I tested it and works just fine.
You can run ecode with `-ldebug` CLI parameter for more detailed logs. Check if the binary is properly detected from ecode with the "Check languages health" tool.

**Reply by ghost** [reply]
I'm using the AppImage, how can I pass `-ldebug`? Btw, language health reports that it found `clangd`.

**Reply by ghost** [reply]
`ecode.log` (without `-ldebug`) shows no signs of the lsp server being invoked.

**Reply by SpartanJ** [reply]
It behaves as any other binary, you can just:
`./ecode-linux-0.6.0-x86_64.AppImage -ldebug`

**Reply by ghost** [reply]
I found a weird bug: the language server is properly started and code completion working normal if I run the AppImage in terminal with `-ldebug`. If I simply click into the AppImage from the file manager then the language server is not invoked and code completion doesn't work.

**Reply by ghost** [reply]
Update: It will work if I run the AppImage in the terminal even without `-ldebug`.

**Reply by SpartanJ** [reply]
This usually happens because they use different PATH.

**Reply by ghost** [reply]
I set `PATH` in `.bashrc`. This could be the reason. But why language health reports that it found `clangd`?

**Reply by ghost** [reply]
Update: I didn't notice the color. It's red so `clangd` is not found!

**Reply by SpartanJ** [reply, marked as answer]
GUI programs do not inherit your shell PATH. This is a pretty common issue.
"Check languages health" gives you all the detail if you click the language row.
Algo you can run `./ecode --health` from the CLI and it will give you the detail from the CLI perspective.

**Reply by SpartanJ** [reply]
Full detail from a lang also can be consulted with: `./ecode --health-lang=cpp`

**Comment by ghost** [comment]
I'm back to FreeBSD. Please continue to provide FreeBSD builds. Thank you.
